internal/kubeclient/indexer.go

```diff
@@ -253,3 +258,53 @@ func indexStagesByWarehouse(obj client.Object) []string {
 	}
 	return nil
 }
+
+func IndexServiceAccountsByRBACGroups(ctx context.Context, mgr ctrl.Manager) error {
+	return mgr.GetFieldIndexer().IndexField(
+		ctx,
+		&corev1.ServiceAccount{},
+		ServiceAccountsByGroupIndexField,
+		indexServiceAccountsByRBACGroups,
+	)
+}
+
+func indexServiceAccountsByRBACGroups(obj client.Object) []string {
+	sa := obj.(*corev1.ServiceAccount) // nolint: forcetypeassert
+	rawGroups := strings.TrimSpace(sa.GetAnnotations()[kargoapi.AnnotationKeyRBACGroups])
+	if rawGroups == "" {
+		return nil
+	}
+	groups := strings.Split(rawGroups, ",")
+	refinedGroups := make([]string, 0, len(groups))
+	for _, g := range groups {
+		if group := strings.TrimSpace(g); group != "" {
+			refinedGroups = append(refinedGroups, group)
+		}
+	}
+	return refinedGroups
+}
```

Comment:
Had a thought about all the new indexing functions...

Should we be validating that the SA's namespace is indeed a Kargo project?

We either have to do that when building the index _or_ after using the index.

I think it's cheaper to do it when building the index, because this function will fire only when a SA is modified and I think that should happen a lot less frequently than when we _use_ the indices during the authn process.